base/src/main/java/org/aya/core/pat/Pat.java

```diff
@@ -225,6 +227,50 @@ record End(boolean isRight, boolean explicit) implements Pat {
     }
   }
 
+  /** TODO[literal]: literal type needs meta-solving for first-class patterns */
```

Comment:
Can you explain this? I'm curious

Reply:
If we have
```
pattern def pat => 0
```
We may solve the result type of `pat` regarding where `pat` is used.

Reply:
Meta should be generated per-use